bundles/org.palladiosimulator.dataflow.confidentiality.analysis/src/org/palladiosimulator/dataflow/confidentiality/analysis/characteristics/variable/PCMDataCharacteristicsCalculator.java

```diff
@@ -28,19 +26,22 @@
 
 import de.uka.ipd.sdq.stoex.AbstractNamedReference;
 
-public class PCMDataCharacteristicsCalculator {
-	private static final Logger logger = Logger.getLogger(PCMDataCharacteristicsCalculator.class);
-    private List<DataFlowVariable> currentVariables;
+public class PCMDataCharacteristicsCalculator implements DataCharacteristicsCalculator {
+    private final List<DataFlowVariable> currentVariables;
+    private final ResourceLoader resourceLoader;
 
     /**
-     * Initialize Characteristic Calculator with initial variables.
+     * Initialize Data characteristics Calculator with initial variables.
      * In addition the read-only container for node characteristics is created. See {@link createNodeCharacteristicsContainer}
      * 
      * @param initialVariables
      *            DataFlowVariables of the previous ActionSequence Element
      */
-    public PCMDataCharacteristicsCalculator(List<DataFlowVariable> initialVariables, List<CharacteristicValue> nodeCharacteristics) {
+    public PCMDataCharacteristicsCalculator(List<DataFlowVariable> initialVariables, 
+    		List<CharacteristicValue> nodeCharacteristics,
```

Comment:
I don't understand the use of node characteristics in this class, we should discuss this (and the javadoc should reflect the usage of the variable)

Reply:
Document that this is because of `.container`